src/test/groovy/ee/tuleva/onboarding/comparisons/fundvalue/retrieval/UnionStockIndexRetrieverSpec.groovy

```diff
@@ -8,10 +8,45 @@ import java.time.LocalDate
 class UnionStockIndexRetrieverSpec extends Specification {
 
     def key = UnionStockIndexRetriever.KEY
-    def fundValueRepository = Mock(FundValueRepository)
-    def retriever = new UnionStockIndexRetriever(fundValueRepository)
+    FundValueRepository fundValueRepository
+    UnionStockIndexRetriever retriever
 
     void setup() {
+        fundValueRepository = Mock(FundValueRepository)
+        retriever = new UnionStockIndexRetriever(fundValueRepository)
+    }
+
+    def "it is configured for the right fund"() {
+        when:
+        def retrievalFund = retriever.getKey()
+
+        then:
+        retrievalFund == UnionStockIndexRetriever.KEY
+    }
+
+    def "it should work with empty data files"() {
```

Comment:
should work with empty data files?

Reply:
empty data files changed to missing index values